generators/controller/templates/app/src/main/java/exception/NotFoundException.java

```diff
@@ -0,0 +1,26 @@
+package <%= packageName %>.exception;
+
+import java.net.URI;
+import java.time.Instant;
+import java.util.List;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ProblemDetail;
+import org.springframework.web.ErrorResponseException;
+
+public class <%= entityName %>NotFoundException extends ErrorResponseException {
+
+    public <%= entityName %>NotFoundException(Long id) {
+        super(HttpStatus.NOT_FOUND, asProblemDetail(id), null);
+    }
+
+    private static ProblemDetail asProblemDetail(Long id) {
```

Comment:
Wouldn't it be better to have exception handling (converting into ProblemDetails) in GlobalExceptionHandler instead of in the Exception class itself? 

Let's say someone is not happy with Spring's ProblemDetails approach and wants to use some other library then they can change that in one place(GlobalExceptionHandler).

Reply:
Makes Sense, I will update it by end of this week.

Reply:
Hi @sivaprasadreddy , to handle this I am thinking of creating a generic NotFoundException which is a RunTimeException and Each Individual Entity will extend it. 

In GlobalExceptionHandler will handle Exception where we can check the instance and based on it set the HttpStatus.

Do you have any other ideas?

Reply:
This looks fine. We can name the base class as `ResourceNotFoundException`.

Reply:
@sivaprasadreddy, Implemented, please review